src/Microsoft.Azure.WebJobs.Host/DefaultParameterResolver.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// The default <see cref="ParameterResolver"/> used by the framework.
+    /// </summary>
+    public class DefaultParameterResolver : ParameterResolver
+    {
+        /// <inheritdoc/>
+        public override bool TryResolve(ParameterResolverContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            string value;
+            BindingParameterResolver resolver = null;
+            if (context.BindingData != null && context.BindingData.TryGetValue(context.ParameterName, out value))
+            {
+                // parameter is resolved from binding data
+                context.Value = value;
+                return true;
+            }
+            else if (BindingParameterResolver.TryGetResolver(context.ParameterName, out resolver))
```

Comment:
Yeah, that's a second level of refactoring that can happen. I could do that in a follow up since its all internal. It's interesting that I had already created an internal "parameter resolver" hierarchy for this stuff :) Good that we're finishing this and making it public. I'll unify this stuff.